### PR TITLE
Add compile-c-module and from-c

### PR DESCRIPTION
**Pull request by LeifAndersen**
(Lacking documentation and tests.)

### CODE REVIEW

File: main.rkt [hunk 1]
```diff
@@ -54,4 +55,7 @@
 
           [zo->compiled-expression (-> zo? compiled-expression?)]
           ;; Compile a zo struct
-))
+
+          [compile-c-module (-> (or/c path-string? path?) void?)]
+          ;; Compile a C module os it can be required in racket
```
Comment (bennn) [inline main.rkt]: "os" -> "so"

**Comment by bennn** [comment]
Cool, can you add docs (not sure how to test this) and fix the dependency problem?
(Travis says: missing `dynext-lib`)

**Comment by LeifAndersen** [comment]
All right, documentation added.

File: scribblings/api.scrbl [hunk 2]
```diff
@@ -17,11 +17,8 @@
 @local-table-of-contents[]
 
 These functions support the REPL, but may be useful in more general settings.
-Import them with `(require zordoz)`.
-Typed variants of the same functions are available with `(require zordoz/typed)`.@note{Require @racket[zordoz/typed/zo-structs] for a typed version of Racket's @racket[compiler/zo-structs]}
```
Comment (bennn) [inline scribblings/api.scrbl]: Keep this line pls.

File: scribblings/api.scrbl [hunk 3]
```diff
@@ -17,11 +17,8 @@
...
+Import them with @racket[(require zordoz)].
 
-@defmodule[zordoz/typed]
-@defmodule[zordoz/typed/zo-structs]
```
Comment (bennn) [inline scribblings/api.scrbl]: Keep these please.

File: scribblings/api.scrbl [hunk 4]
```diff
@@ -205,3 +207,56 @@ Tools for compiling syntax fragments rather than entire modules.
...
 ]
+
+@section{Compiling C Modules}
+
+Tools for compiling modules implemented in C.
+
+@defproc[(compile-c-module [c-path (or/c path-string? path?)]) void?]{
+Compiles a C module to a form where it can be required later.
+
+@larger{@larger{@larger{WARNING:}}}
```
Comment (bennn) [inline scribblings/api.scrbl]: That's really big. Can we replace 2 `@larger` with 1 `@bold`?

**Pull request merged**